### DISCUSSION TITLE
What is Difference between useEffect vs useLayoutEffect in React.js

### DISCUSSION THREAD

**Discussion by strong-728** [community/community#35135] [Other Feature Feedback, Questions, & Ideas]
Please help me, urgently must

**Comment by codeshare007** [comment]
The difference between them is the timing of their invocation.
useEffect runs asynchronously after a render is painted to the screen, unblocking the browser paint process.
So that looks like:
1. We cause a render somehow (through the state, or the parent re-renders or context change). In our case by clicking on the Increment button.
2. React updates the state internally. Counter value gets updated to 1.
3. React handles the DOM mutation. <h1>Count: 0 </h1> gets changed to <h1>Count: 1 </h1>
4. The browser paints this DOM change to the browser’s screen. In our case Count: 1 is painted on the screen.
5. The useEffect function is fired after the browser has painted the DOM changes. Prints useEffect is fired.
However, useLayoutEffect fires synchronously after all DOM mutations.
So that looks like:
1. We cause a render somehow (through the state, or the parent re-renders or context change). In our case by clicking on the Increment button.
2. React updates the state internally. Counter value gets updated to 1.
3. React handles the DOM mutation. <h1>Count: 0 </h1> gets changed to <h1>Count: 1 </h1>
4. useLayoutEffect is fired immediately after the DOM mutations.
5. The browser paints this DOM change to the browser’s screen. In our case Count: 1 is painted on the screen.

In simple words, useLayoutEffect doesn’t care whether the browser has painted the DOM changes or not. It triggers the function right after the DOM mutations are computed.

**Comment by ghost** [comment]
The useLayoutEffect function is triggered synchronously before the DOM mutations are painted. However, the useEffect function is called after the DOM mutations are painted.
